pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -601,6 +601,9 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusterForPortOrUDS(pluginPara
 	localCluster := buildDefaultCluster(pluginParams.Env, clusterName, apiv2.Cluster_STATIC, localityLbEndpoints,
 		model.TrafficDirectionInbound, pluginParams.Node, nil)
 	setUpstreamProtocol(localCluster, instance.Endpoint.ServicePort)
+	// Set the upstream protocol selection based on the downstream protocol. This is for inbound clusters only.
+	// Use HTTP1.1 or HTTP2, depending on which one is used on the downstream connection.
+	localCluster.ProtocolSelection = apiv2.Cluster_USE_DOWNSTREAM_PROTOCOL
```

Comment:
what happens in this scenario ?
```
clientProxy - ( h2 upgrade ) ---->  serverProxy --(inbound cluster) --->  (http 1.1) application```
It seems that serverProxy will forward the h2 connection to the application which cannot speak h2. we need to guard for that.

Reply:
Envoy does not support h2 upgrades in inbound or in cluster. So we don't have to worry about that problem. Secondly, guarding for this specific condition means we break all the protocol sniffing work where the application does not declare the port type. I think the latter is a much bigger win compared to the former. The former will be solved when envoy gains proper h2 upgrade facility.